### ISSUE TITLE
When a user quit the channel, it does not store the channel

### ISSUE THREAD

**Issue by AntoineTurmel** [classam/pierc#37]
When a user quit the channel, it does not store the channel in the main table.
"undefined" is put instead of the right channel name:

![bug_pierc](https://cloud.githubusercontent.com/assets/119288/9256805/16c1a792-41f2-11e5-99a7-9f8cb26d7aa6.png)

**Comment by frdmn** [comment]
I assume this is intended because the bot actually doesn't listen for "leaving a channel" (`PART`) event, instead it listens for the `QUIT` event, which doesn't reference a specific channel because the user disconnects from the IRC server completely.

**Comment by AntoineTurmel** [comment]
It's not possible to write a quit line for each channel where the user was ?

**Comment by frdmn** [comment]
No, not really, as you can see [here](https://github.com/classam/pierc/blob/master/bot/pierc.py#L83-L87), the bot tries to parse the channel if the event contains one. However if the user quits instead of parts the channel, the bot won't be able to tell which channels the person was participating.

**Comment by AntoineTurmel** [comment]
Well the bot can determine the quit is from X channel because the quit message is displayed on the channel no ?

**Comment by frdmn** [comment]
I don't think that's how the IRC library works. In general it receives events, those can be messages, join events, quit events, topic change events, etc. At the moment, pierc just receives those and writes it's information into the database. However, the `quit` event doesn't contain a channel or target which we can parse. At this point we don't know "where" the quit came frome.

**Comment by AntoineTurmel** [comment]
Ok, so I suppose we need first to detect if an user is in the channel... and then deduce by the quit that he is no more in the channel.
